### PR TITLE
Migrate jquery

### LINKED ISSUE #1155: migrate jquery
A long time ago we added jquery-migrate to update jquery in place without actually migrating. Now, years later, I think is the time to _actually_ migrate jquery.

I don't know all the spots where we need to do this in, but it seems like we should start to identify them and slowly migrate what we can until (a) we have no more deprecation warnings and (b) we can get rid of jquery-migrate altogether and just use jquery 3 directly.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201627247816969) by [Unito](https://www.unito.io)

### LINKED ISSUE #1516: migrate application.js to webpack
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201627543540123) by [Unito](https://www.unito.io)

### LINKED ISSUE #1783: fix data-confirm-modal
#1782 broke the `data-confirm-modal` functionality from a gem. There's an npm package we can use. It needs `@rails/ujs`, so we'll need to look into what that entails.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201712338600894) by [Unito](https://www.unito.io)

### LINKED ISSUE #1784: migrate sprocket require chain
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201712341939618) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by johrstrom**
This migrate jquery to webpack so that we can then remove the migration. Related to #1005.

I tried to make this as small as I could, but once one thing broke - the whole chain started to go.  This removes everything from sprockets so a lot of the smaller updates are around making sure jquery is available everywhere, even to inline scripts.

Fixes #1784, even though I didn't necessarily want to.
Fixes #1155, even though I didn't necessarily want to.
Fixes #1783 (even though it created it).

This removes fontawesome-iconpicker, which I want to re-implement because it has redifinitions of `popover` among other things like it being kinda unmaintainable lookup tables and minified variables.

This needs testing all sorts of places

* activejobs
* products indexes
* files app

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201713122224591) by [Unito](https://www.unito.io)

### CODE REVIEW

File: apps/dashboard/app/javascript/packs/active_jobs.js [hunk 1]
```diff
@@ -1,11 +1,10 @@
+'use strict';
 
 import oboe from 'oboe';
+import 'datatables.net';
+import 'datatables.net-bs4/js/dataTables.bootstrap4';
 import 'datatables.net-plugins/api/processing()';
 
-jQuery(function(){
-    $('[data-toggle="tooltip"]').tooltip();
-});
```
Comment (johrstrom) [inline apps/dashboard/app/javascript/packs/active_jobs.js:7]: this is taken care of in application.js

**Comment by johrstrom** [comment]
This is ready for review. It's going into a feature branch until #1005 is completely done because this does break icon picker directly and breaks our modals.

**Comment by johrstrom** [comment]
OK - I fixed modals. And I think the `@rails/ujs` is unavoidable as data-confirm-modal needs and it we use data-confirm-modal.

**Comment by johrstrom** [comment]
This is ready for review now. Could have some bugs. I'd say - test all pages and watch the console logs for errors.

**Comment by johrstrom** [comment]
Switching this into master because I have a replacement for the iconpicker (which is currently broken anyway), just without search.

**Comment by gbsoftwaresolutions** [comment]
In Active Jobs, when expanding a job by clicking on the "Arrow".  All looks good except for the "Detailed Metrics".  We show "Bad Request" in that section.  When I click on the link "Detailed Metrics", I am taken to a url that gives the same message.

image

**Comment by gbsoftwaresolutions** [comment]
Products indexes.

image


Bundle Install, Restart App, 
* products_show.js:56 Uncaught TypeError: $(...).modal is not a function
    at updateModal (products_show.js:56:1)
    at HTMLButtonElement.<anonymous> (products_show.js:9:1)
    at HTMLButtonElement.dispatch (jquery.js:4670:1)
    at HTMLButtonElement.elemData.handle (jquery.js:4490:1)


Shell and Files Works fine.
Edit MetaData.  The only problem is the iconpicker, which you already explained

**Comment by gbsoftwaresolutions** [comment]
> In Active Jobs, when expanding a job by clicking on the "Arrow". All looks good except for the "Detailed Metrics". We show "Bad Request" in that section. When I click on the link "Detailed Metrics", I am taken to a url that gives the same message.
> 
> image



if i change the grana-test url to grafana url, the error goes away. this is specific to test.

**Comment by gbsoftwaresolutions** [comment]
> Products indexes.
> 
> image
> 
> Bundle Install, Restart App,
> 
> * products_show.js:56 Uncaught TypeError: $(...).modal is not a function
>   at updateModal (products_show.js:56:1)
>   at HTMLButtonElement. (products_show.js:9:1)
>   at HTMLButtonElement.dispatch (jquery.js:4670:1)
>   at HTMLButtonElement.elemData.handle (jquery.js:4490:1)
> 
> Shell and Files Works fine. Edit MetaData. The only problem is the iconpicker, which you already explained

if I use prod dashboard against dev products, this works fine.

**Comment by johrstrom** [comment]
> if i change the grana-test url to grafana url, the error goes away. this is specific to test.

You need to clear your cookies for that site. The response is `size of a request header field exceeds server limit` which means it's your OIDC cookies piling up.

**Review by gbsoftwaresolutions — COMMENTED** [review]
apps/dashboard/app/assets/javascripts/application.js 

the file is empty. We should probably delete it, unless it's existence is needed to prevent errors.

**Comment by johrstrom** [comment]
> the file is empty. We should probably delete it, unless its existence is needed to prevent errors.

You're right on. sprockets is still in the toolchain at this point and it expects a file there.

> products_show.js:56 Uncaught TypeError: $(...).modal is not a function

I think I fixed that but I'm going to right some test cases around the same.

**Comment by gbsoftwaresolutions** [comment]
> > the file is empty. We should probably delete it, unless its existence is needed to prevent errors.
> 
> You're right on. sprockets is still in the toolchain at this point and it expects a file there.
> 
> > products_show.js:56 Uncaught TypeError: $(...).modal is not a function
> 
> I think I fixed that but I'm going to right some test cases around the same.

Hey @johrstrom This functionality is good.  However, I am seeing the following error at the bottom of the page.
I have tried the following in the dashboard folder, to fix but to no avail.
* bundle install
* gem install rails_12factor
* bin/setup
* bundle install
* bin/setup


Please fix the errors below:

Gemfile missing rails_12factor gem
Not a valid git repo

**Review by gbsoftwaresolutions — APPROVED** [review]
All changes appear to be testing successfully.

**Pull request merged**